go/pkg/tree/tree.go

```diff
@@ -263,6 +262,14 @@ func flattenTree(root digest.Digest, rootPath string, dirs map[digest.Digest]*re
 			return nil, fmt.Errorf("couldn't find directory %s with digest %s", flatDir.p, flatDir.d)
 		}
 
+		// Check whether this is an empty directory.
+		if len(dir.Files) == 0 && len(dir.Directories) == 0 && len(dir.Symlinks) == 0 {
```

Comment:
You can add the three and do a single comparison check ;)

Reply:
Oh, right :-)